ignore ts errors

mifi opened the pull request:
i couldn't get the @types/node-redis-pubsub to work
It seems they are incorrect (they have a namespace NRP around teh class) (NRP.NodeRedisPubSub)
See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node-redis-pubsub/index.d.ts
I'm, not so good with TS but this seems wrong. If we change the code to match the TS definititions, I think it's gonna break

goto-bus-stop reviewed (commented):
I think they did it this way since you can construct an NRP instance with or without `new` in JS. But their strategy breaks extending it in a class in Typescript.

I _think_ we can try a different approach to address it, by casting the imported type to `typeof NRP.NodeRedisPubSub`. Then we still get type checking.

```js
/** @type {typeof import('node-redis-pubsub').NodeRedisPubSub} */
// @ts-ignore the node-redis-pubsub types are off, unsafe cast to the actual type
const NRP = require('node-redis-pubsub')
```

Not entirely sure if TS lets us do that, but it's worth a shot?

goto-bus-stop commented:
If that _doesn't_ work, then this PR is the next best thing :)

mifi commented:
I tried with the ts-ignore. it would then complain that `this.on` is not a method etc. And typeof is not correct because we don't want the type of the type (syntax error). if you think you can make the typings works then please do. but i tried to do some googling and from what i found, if some types are incorrect, there's not much we can do other than submit a pr to fix it or patch it.

goto-bus-stop commented:
OK, I'm fine with the approach in this PR or with getting rid of the inheritance, whichever you'd rather do :)

mifi commented:
I have now converted it from inherits class to factory function.

I also added a commit that fixes a critical bug introduced in the chunkSize PR ( https://github.com/transloadit/uppy/pull/2881 )
I guess ideally it should be a separate pr but we might as well include it here?

arturi commented:
Looks good to merge?
